core/src/test/java/org/mskcc/oncotree/utils/VersionUtilTest.java

```diff
@@ -0,0 +1,271 @@
+                    failureReport.append(versionName + " : version expected but not returned from versionUtil\n");
+                }
+            } else {
+                String expectedVersionName = expectedVersion.getVersion();
+                String returnedVersionName = returnedVersion.getVersion();
+                if (!testValuesMatch(expectedVersionName, returnedVersionName)) {
+                    failureReport.append(makeMismatchMessage(versionName, "version", returnedVersionName, expectedVersionName));
+                    failureCount = failureCount + 1;
+                }
+                String expectedGraphURI = expectedVersion.getGraphURI();
+                String returnedGraphURI = returnedVersion.getGraphURI();
+                if (!testValuesMatch(expectedGraphURI, returnedGraphURI)) {
+                    failureReport.append(makeMismatchMessage(versionName, "graphURI", returnedGraphURI, expectedGraphURI));
+                    failureCount = failureCount + 1;
+                }
+                String expectedDescription = expectedVersion.getDescription();
+                String returnedDescription = returnedVersion.getDescription();
+                if (!testValuesMatch(expectedDescription, returnedDescription)) {
+                    failureReport.append(makeMismatchMessage(versionName, "description", returnedDescription, expectedDescription));
+                    failureCount = failureCount + 1;
+                }
+            }
+        }
+        if (expectedVersionMap.size() != returnedVersionMap.size()) {
+            failureReport.append("mismatch between returned version list (size:" + Integer.toString(returnedVersionMap.size()) +
+                    ") and expected version list (size:" + Integer.toString(expectedVersionMap.size()) + ")");
+            failureCount = failureCount + 1;
+        }
+        if (failureCount > 0) {
+            fail(Integer.toString(failureCount) + " failed test conditions. Details follow... " + failureReport.toString());
+        }
+        return;
+    }
+
+    @Test
+    public void testSuccessfulVersionQuery() throws Exception {
+        setupForTest();
+        String testVersionName = expectedVersionMap.keySet().iterator().next();
+        Version expectedVersion = expectedVersionMap.get(testVersionName);
+        Version returnedVersion = versionUtil.getVersion(testVersionName);
+        int failureCount = 0;
+        StringBuilder failureReport  = new StringBuilder();
+        if (expectedVersion == null || returnedVersion == null) {
+            if (expectedVersion == null) {
+                failureReport.append(testVersionName + " : no values in expected version hash (internal error)\n");
+            }
+            if (returnedVersion == null) {
+                failureReport.append(testVersionName + " : version expected but not returned from versionUtil\n");
+            }
+        } else {
+            String expectedVersionName = expectedVersion.getVersion();
```

Comment:
Is all this code exactly the same as above? If so maybe it can be it's own method to minimize code duplication.